### PR TITLE
CASSANDRA-19502: ensure that tests do not overwrite cassandra-topology.properties

### PR DESCRIPTION
**Pull request by Claudenw**
Thanks for sending a pull request! Here are some tips if you're new here:
 
 * Ensure you have added or run the [appropriate tests](https://cassandra.apache.org/_/development/testing.html) for your PR.
 * Be sure to keep the PR description updated to reflect all changes.
 * Write your PR title to summarize what this PR proposes.
 * If possible, provide a concise example to reproduce the issue for a faster review.
 * Read our [contributor guidelines](https://cassandra.apache.org/_/development/index.html)
 * If you're making a documentation change, see our [guide to documentation contribution](https://cassandra.apache.org/_/development/documentation.html)
 
Commit messages should follow the following format:

```
<One sentence description, usually Jira title or CHANGES.txt summary>

<Optional lengthier description (context on patch)>

patch by <Authors>; reviewed by <Reviewers> for CASSANDRA-#####

Co-authored-by: Name1 <email1>
Co-authored-by: Name2 <email2>

```

The [Cassandra Jira](https://issues.apache.org/jira/projects/CASSANDRA/issues/)

### CODE REVIEW

**Comment by Claudenw** [comment]
@michaelsembwever, If you recall I submitted a pull request for Cassandra-12937 that included a change to the `topology.properties` file.  You asked why that change was made and I could not answer as I received it from a development branch that I did not work on.  However, I discovered that if you run `ant test` that file would be overwritten.  While this is not a problem on the CI system it does cause a problem on development systems where the modified file can be accidentally checked in :raised_hand_with_fingers_splayed:.  This change to one file solves the problem.  Please review and let's see if we can get this in.

File: test/unit/org/apache/cassandra/locator/PropertyFileSnitchTest.java [hunk 1]
```diff
@@ -72,66 +72,95 @@ public void setup() throws ConfigurationException, IOException
         backupFile = Paths.get(confFile + ".bak");
         localAddress = FBUtilities.getBroadcastAddressAndPort();
 
-        restoreOrigConfigFile();
+        Files.copy(effectiveFile, backupFile, java.nio.file.StandardCopyOption.REPLACE_EXISTING);
```
Comment (Mmuzaf) [inline test/unit/org/apache/cassandra/locator/PropertyFileSnitchTest.java:76]: `@After` should work as well, right?

Comment (Claudenw) [reply test/unit/org/apache/cassandra/locator/PropertyFileSnitchTest.java:76]: Not for the `Files.copy` but for the `finally` of the try/catch that calls `restoreOrigConfigFile()`.  In fact `restoreOrigConfigFile` could be annotated with `@After`

Comment (Mmuzaf) [reply test/unit/org/apache/cassandra/locator/PropertyFileSnitchTest.java:76]: I think this is the right way to go :-)

Comment (Claudenw) [reply test/unit/org/apache/cassandra/locator/PropertyFileSnitchTest.java:76]: Updated pull request after testing to ensure that it works as expected.

**Comment by Mmuzaf** [comment]
I can confirm that the problem is reproducible without the changes, and is fixed with the changes.

The command to reproduce:
`ant test` (the scope of the files must be narrowed in advance)

However, the command below doesn't reproduce the issue: 
`ant testclasslist -Dtest.classlistfile=testnameslist.txt`

**Review by Mmuzaf — APPROVED** [review]

**Pull request closed without merging**